lua/autoclose.lua

```diff
@@ -67,7 +67,12 @@ local function is_disabled(info)
    if config.disabled then
       return true
    end
-   local current_filetype = vim.api.nvim_buf_get_option(0, "filetype")
+   local current_filetype
+   if vim.fn.has("nvim-0.10.0") == 1 then
+      current_filetype = vim.api.nvim_get_option_value("filetype", {buf = 0})
+   else
+      current_filetype = vim.api.nvim_buf_get_option(0, "filetype")
+   end
```

Comment:
```suggestion
local current_filetype = vim.fn.has("nvim-0.10.0") == 1 
    and vim.api.nvim_get_option_value("filetype", {buf = 0})
    or vim.api.nvim_buf_get_option(0, "filetype")
```

Reply:
For whatever it's worth, I prefer the original version because it's simpler and is easier to read than the `==...and...or` chain.

Reply:
I think this approach makes the code more concise and avoids explicit `if` statements.

Reply:
I agree that it's more concise, but shorter is not always better. I also don't think that code needs to (or should) "avoid explicit `if` statements."

Bottom line, I think that the version with `if` is simpler to read (and thus will be better going forward).